### PR TITLE
Add publication time to registry package release metadata

### PR DESCRIPTION
**Pull request by yim-lee**
Motivation:
Registry should have information about how a package was published. This includes who published it, who signed it, and when it was published.
 - "Who published the package release?": this is `metadata.author`, which is already defined in the metadata JSON schema.
 - "Who signed the package release?": this can be extracted from the signing certificate if signed (requires rdar://107483043 and will be done in separate PR)
 - "When was the release published?": added in this PR

rdar://106674475

Modifications:
- Add `originalPublicationTime` to user-provided metadata during publication
- Add `publishedAt` to registry's package release metadata response
- Update registry API spec

### CODE REVIEW

File: Documentation/Registry.md [hunk 1]
```diff
@@ -1706,6 +1710,135 @@ components:
...
+</details>
+
+##### `PackageRelease` type
+
+| Property                  | Type                | Description                                      | Required |
+| ------------------------- | :-----------------: | ------------------------------------------------ | :------: |
+| `author`                  | [Author](#author-type) | Author of the package release. | |
+| `description`             | String | A description of the package release. | |
+| `licenseURL`              | String | URL of the package release's license document. | |
+| `originalPublicationTime` | String | Original publication time of the package release in [ISO 8601] format. This can be set if the package release was previously published elsewhere.<br>A registry should record the publication time independently and include it as `publishedAt` in the [package release metadata response](#42-fetch-information-about-a-package-release). <br>In case both `originalPublicationTime` and `publishedAt` are set, `originalPublicationTime` should be used. | |
```
Comment (yim-lee) [inline Documentation/Registry.md:1819]: This is new

File: Documentation/Registry.md [hunk 2]
```diff
@@ -466,18 +467,20 @@ Link: <https://packages.example.com/mona/LinkedList/1.1.1>; rel="latest-version"
...
-| `version`   | String | The package release version number.       |
-| `resources` | Array  | The resources available for the release.  |
-| `metadata`  | Object | Additional information about the release. |
+| Key           | Type   | Description                               | Required |
+| ------------- | ------ | ----------------------------------------- | :------: |
+| `id`          | String | The namespaced package identifier.        | ✓ |
+| `version`     | String | The package release version number.       | ✓ |
+| `resources`   | Array  | The resources available for the release.  | ✓ |
+| `metadata`    | Object | Additional information about the release. | ✓ |
+| `publishedAt` | String | The [ISO 8601]-formatted datetime string of when the package release was published, as recorded by the registry. See related [`originalPublicationTime`](#appendix-b---package-release-metadata-json-schema) in `metadata`. | |
```
Comment (yim-lee) [inline Documentation/Registry.md:483]: This is new

**Comment by yim-lee** [comment]
@swift-ci please smoke test

**Comment by yim-lee** [comment]
@swift-ci please smoke test Linux

**Comment by yim-lee** [comment]
@swift-ci please smoke test Linux

**Comment by neonichu** [comment]
Can we also add this to `Sources/PackageMetadata/PackageMetadata.swift`?

**Comment by yim-lee** [comment]
@neonichu Added to `PackageMetadata.swift`: https://github.com/apple/swift-package-manager/pull/6407/commits/73990fb43898a04ea0f91875f3e1adf541f69c20

@swift-ci please smoke test

**Comment by yim-lee** [comment]
@swift-ci please test Windows platform

File: Sources/PackageMetadata/PackageMetadata.swift [hunk 3]
```diff
@@ -100,20 +117,23 @@ public struct PackageSearchClient {
...
     }
 
     var repositoryProvider: RepositoryProvider {
-        return GitRepositoryProvider()
+        GitRepositoryProvider()
     }
 
     // FIXME: This matches the current implementation, but we may want be smarter about it?
     private func guessReadMeURL(baseURL: URL, defaultBranch: String) -> URL {
-        return baseURL.appendingPathComponent("raw").appendingPathComponent(defaultBranch).appendingPathComponent("README.md")
```
Comment (neonichu) [inline Sources/PackageMetadata/PackageMetadata.swift:114]: 😿

**Review by tomerd — APPROVED** [review]

**Pull request merged**